src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,240 @@
+                DateTimeFormatter fromFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                try {
+                    LocalDate.parse(by, fromFormat);
+                } catch (DateTimeParseException e) {
+                    System.out.println("Deadline format has to be in yyyy-MM-dd");
+                    return;
+                }
+                Deadline l = new Deadline(description, LocalDate.parse(by));
+                lst.add(l);
+                System.out.println("Got it. I've added this task:\n " + l +
+                        "\nNow you have " + lst.count() + " tasks in the list.");
+                storage.updateFile(lst);
+            } else {
+                throw new DukeException("You have to include the deadline");
+            }
+        } else {
+            throw new DukeException("The description of a deadline cannot be empty.");
+        }
+    }
+
+    /**
+     * Adds a Event to the tasklist
+     *
+     * @param str The users input
+     * @param lst The tasklist
+     * @param storage the storage object
+     */
+    public static void eventMessage(String str, TaskList lst, Storage storage) throws DukeException, IOException {
+        if (str.length() > 6) {
+            int sepPos = str.indexOf("/");
+            if (sepPos != -1) {
+                String description = str.substring(6, sepPos);
+                String at = str.substring(sepPos + 4);
+                Event e = new Event(description, at);
+                lst.add(e);
+                System.out.println("Got it. I've added this task:\n " + e +
+                        "\nNow you have " + lst.count() + " tasks in the list.");
+                storage.updateFile(lst);
+            } else {
+                throw new DukeException("You have to include the timings of an event");
+            }
+        } else {
+            throw new DukeException("The description of a event cannot be empty.");
+        }
+    }
+
+    /**
+     * Deletes a task from the tasklist
+     *
+     * @param str The users input
+     * @param lst The tasklist
+     * @param storage the storage object
+     */
+    public static void deleteMessage(String str, TaskList lst, Storage storage) throws DukeException, IOException {
+        if (str.length() > 7) {
+            char index = str.charAt(7);
+            int number = Integer.parseInt(String.valueOf(index));
+            if(number > lst.count()) {
+                throw new DukeException("No such task.");
+            }
+            Task t = lst.get(number - 1);
+            lst.delete(t);
+            System.out.println("Noted. I've removed this task: \n " + t + "\nNow you have "
+                    + lst.count() + " tasks in the list.");
+            storage.updateFile(lst);
+        }
+    }
+
+    /**
+     * Finds a task based on a keyword
+     *
+     * @param str The users input
+     * @param lst The tasklist
+     * @param storage the storage object
+     */
```

Comment:
Good that all public methods have JavaDoc comments

Reply:
Thanks!